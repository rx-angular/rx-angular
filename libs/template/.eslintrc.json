{
  "extends": "../../.eslintrc.json",
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts"],
      "extends": [
        "plugin:@nrwl/nx/angular",
        "plugin:@angular-eslint/template/process-inline-templates"
      ],
      "rules": {
        /**
         * @Notice: overwrite root enforce-module-boundaries to allow deep imports of secondary entry-points.
         */
        "@nrwl/nx/enforce-module-boundaries": [
          "error",
          {
            "enforceBuildableLibDependency": false,
            "allow": ["@rx-angular/template/**"],
            "depConstraints": [
              {
                "sourceTag": "template",
                "onlyDependOnLibsWithTags": ["type:test-util", "cdk"]
              }
            ]
          }
        ],
        "@angular-eslint/directive-selector": [
          "error",
          {
            "type": "attribute",
            "prefix": "rx",
            "style": "camelCase"
          }
        ],
        "@angular-eslint/component-selector": [
          "error",
          {
            "type": "element",
            "prefix": "rx",
            "style": "kebab-case"
          }
        ],
        "@typescript-eslint/no-non-null-assertion": "warn"
      }
    },
    {
      "files": ["*.html"],
      "extends": ["plugin:@nrwl/nx/angular-template"],
      "rules": {
        "@angular-eslint/template/eqeqeq": "warn"
      }
    }
  ]
}
