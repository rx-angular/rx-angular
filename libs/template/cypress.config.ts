import { defineConfig } from 'cypress';
import { nxComponentTestingPreset } from '@nrwl/angular/plugins/component-testing';

export default defineConfig({
  component: {
    ...nxComponentTestingPreset(__filename),
    specPattern: ['**/*.cy.ts'],
    viewportHeight: 600,
  },
  viewportHeight: 600,
});
