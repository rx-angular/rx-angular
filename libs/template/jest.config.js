module.exports = {
  bail: true,
  preset: '../../jest.preset.js',
  coverageReporters: ['lcov', 'cobertura'],
  coverageDirectory: '../../docs/test-coverage/template',
  collectCoverageFrom: [
    './src/**/!(index).ts',
    '!./src/lib/experimental/**/*.ts',
  ],

  setupFilesAfterEnv: ['<rootDir>/src/test-setup.ts'],
  globals: {
    'ts-jest': {
      tsconfig: '<rootDir>/tsconfig.spec.json',
      stringifyContentPathRegex: '\\.(html|svg)$',
    },
  },
  displayName: 'template',
  snapshotSerializers: [
    'jest-preset-angular/build/serializers/no-ng-attributes',
    'jest-preset-angular/build/serializers/ng-snapshot',
    'jest-preset-angular/build/serializers/html-comment',
  ],
  transform: { '^.+\\.(ts|js|html)$': 'jest-preset-angular' },
};
