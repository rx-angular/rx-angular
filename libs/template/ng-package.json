{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../dist/libs/template",
  "lib": {
    "flatModuleFile": "template",
    "entryFile": "src/index.ts",
    "umdModuleIds": {
      "@rx-angular/cdk": "rx-angular.cdk",
      "@rx-angular/cdk/zone-less": "rx-angular.cdk-zone-less",
      "@rx-angular/cdk/zone-configurations": "rx-angular.cdk-zone-configurations"
    }
  }
}
