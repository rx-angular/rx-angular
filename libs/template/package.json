{
  "name": "@rx-angular/template",
  "version": "1.0.0-beta.29",
  "description": "**Fully** Reactive Component Template Rendering in Angular. @rx-angular/template aims to be a reflection of Angular's built in renderings just reactive.",
  "main": "index.js",
  "publishConfig": {
    "access": "public"
  },
  "engines": {
    "node": ">=10.12.0",
    "npm": ">=5.3.0",
    "yarn": ">=1.9.2"
  },
  "author": "Michael Hladky <michael@hladky.at>",
  "homepage": "https://rx-angular.io/",
  "keywords": [
    "angular",
    "rxjs",
    "performance",
    "rendering",
    "template",
    "lazy templates",
    "memoized views",
    "scheduling",
    "coalescing",
    "concurrent rendering"
  ],
  "contributors": [
    {
      "name": "Michael Hladky",
      "email": "michael@hladky.at"
    },
    {
      "name": "Julian Jandl",
      "email": "julian@jandl.wien"
    },
    {
      "name": "Kirill Karnaukhov",
      "email": "karnaukhov.kh@gmail.com"
    },
    {
      "name": "Edouard Bozon",
      "email": "bozonedouard@gmail.com"
    }
  ],
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "https://github.com/rx-angular/rx-angular.git"
  },
  "peerDependencies": {
    "@angular/core": ">10.1.1",
    "@rx-angular/cdk": ">=1.0.0-alpha.11"
  },
  "dependencies": {
    "tslib": "^2.0.0"
  },
  "schematics": "./schematics/collection.json",
  "ng-add": {
    "save": "dependencies"
  },
  "ng-update": {
    "migrations": "./schematics/migration.json"
  }
}
