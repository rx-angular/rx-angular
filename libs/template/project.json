{
  "projectType": "library",
  "root": "libs/template",
  "sourceRoot": "libs/template/src",
  "prefix": "rx",
  "targets": {
    "build-lib": {
      "executor": "@nrwl/angular:package",
      "options": {
        "tsConfig": "libs/template/tsconfig.lib.json",
        "project": "libs/template/ng-package.json"
      }
    },
    "build-schematics": {
      "executor": "@nrwl/workspace:run-commands",
      "options": {
        "commands": [
          {
            "command": "npx tsc -p libs/template/tsconfig.schematics.json"
          },
          {
            "command": "npx cpx libs/template/schematics/collection.json dist/libs/template/schematics"
          },
          {
            "command": "npx cpx libs/template/schematics/migration.json dist/libs/template/schematics"
          },
          {
            "command": "npx cpx libs/template/schematics/src/**/files/** dist/libs/template/schematics/src"
          },
          {
            "command": "npx cpx libs/template/schematics/src/**/schema.json dist/libs/template/schematics/src"
          }
        ],
        "parallel": false
      }
    },
    "build": {
      "executor": "@nrwl/workspace:run-commands",
      "options": {
        "commands": [
          {
            "command": "nx run template:build-lib"
          },
          {
            "command": "nx run template:build-schematics"
          }
        ],
        "parallel": false
      }
    },
    "lint": {
      "executor": "@angular-devkit/build-angular:tslint",
      "options": {
        "tsConfig": [
          "libs/template/tsconfig.lib.json",
          "libs/template/tsconfig.spec.json"
        ],
        "exclude": ["**/node_modules/**", "!libs/template/**/*"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "options": {
        "jestConfig": "libs/template/jest.config.js",
        "passWithNoTests": true
      },
      "outputs": ["coverage/libs/template"]
    },
    "build-docs": {
      "executor": "@nrwl/workspace:run-commands",
      "sequential": true,
      "options": {
        "commands": [
          {
            "command": "nx test-coverage template && nx test-coverage-badge template"
          }
        ]
      }
    },
    "test-coverage-badge": {
      "executor": "@nrwl/workspace:run-commands",
      "options": {
        "commands": [
          {
            "command": "npx istanbul-cobertura-badger -b jest-coverage-badge -r ./docs/test-coverage/template/cobertura-coverage.xml -d docs/test-coverage/template"
          }
        ]
      }
    },
    "test-coverage": {
      "executor": "@nrwl/jest:jest",
      "options": {
        "jestConfig": "libs/template/jest.config.js",
        "passWithNoTests": true,
        "codeCoverage": true
      },
      "outputs": ["coverage/libs/template"]
    },
    "changelog": {
      "executor": "@nrwl/workspace:run-commands",
      "options": {
        "commands": [
          {
            "command": "npx conventional-changelog-cli -p angular -k ./libs/template/package.json -i ./libs/template/CHANGELOG.md -s --commit-path ./libs/template --tagPrefix template@"
          }
        ]
      }
    },
    "link": {
      "executor": "@nrwl/workspace:run-commands",
      "options": {
        "commands": [
          {
            "command": "cd ./dist/libs/template && npm link"
          }
        ]
      }
    },
    "publish": {
      "executor": "@nrwl/workspace:run-commands",
      "options": {
        "commands": [
          {
            "command": "nx test template && nx build template && cd ./dist/libs/template && npm publish"
          }
        ]
      }
    }
  },
  "generators": {
    "@nrwl/angular:component": {
      "style": "scss"
    }
  },
  "tags": ["type:lib", "type:template"]
}
