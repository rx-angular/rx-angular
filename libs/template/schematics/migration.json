{
  "$schema": "../../../node_modules/@angular-devkit/schematics/collection-schema.json",
  "schematics": {
    "update-1-0-0": {
      "version": "1.0.0-beta.29",
      "description": "Migrate to sub-entry points",
      "factory": "./src/migrations/update-1-0-0/index"
    }
  }
}
