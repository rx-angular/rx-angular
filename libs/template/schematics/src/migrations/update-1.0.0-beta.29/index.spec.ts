import { Tree } from '@angular-devkit/schematics';
import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import * as path from 'path';

describe('Template Migration 1.0.0-beta.29', () => {
  let appTree: UnitTestTree;

  it('should replace LetModule + LetDirective module specifier', async () => {
    appTree = await setupTestFile(`
      import { NgModule } from '@angular/core';
      import { BrowserModule } from '@angular/platform-browser';
      import {
        LetModule,
        LetDirective
      } from '@rx-angular/template';

      import { AppComponent } from './app.component';

      @NgModule({
        declarations: [
          AppComponent,
        ],
        imports: [
          BrowserModule,
          LetModule
        ],
        providers: [],
        bootstrap: [AppComponent]
      })
      export class AppModule { }
  `);

    const file = appTree.readContent('app.module.ts');

    expect(file).toMatchSnapshot();
  });

  it('should replace PushModule + PushPipe module specifier', async () => {
    appTree = await setupTestFile(`
      import { NgModule } from '@angular/core';
      import { BrowserModule } from '@angular/platform-browser';
      import { PushModule, PushPipe } from '@rx-angular/template';

      import { AppComponent } from './app.component';

      @NgModule({
        declarations: [
          AppComponent,
        ],
        imports: [
          BrowserModule,
          PushModule
        ],
        providers: [],
        bootstrap: [AppComponent]
      })
      export class AppModule { }
  `);

    const file = appTree.readContent('app.module.ts');

    expect(file).toMatchSnapshot();
  });

  it('should replace UnpatchEventsModule + UnpatchDirective module specifier', async () => {
    appTree = await setupTestFile(`
      import { NgModule } from '@angular/core';
      import { BrowserModule } from '@angular/platform-browser';
      import { UnpatchEventsModule, UnpatchDirective } from '@rx-angular/template';

      import { AppComponent } from './app.component';

      @NgModule({
        declarations: [
          AppComponent,
        ],
        imports: [
          BrowserModule,
          UnpatchEventsModule
        ],
        providers: [],
        bootstrap: [AppComponent]
      })
      export class AppModule { }
  `);

    const file = appTree.readContent('app.module.ts');

    expect(file).toMatchSnapshot();
  });

  it('should replace all module specifiers', async () => {
    appTree = await setupTestFile(`
    import { NgModule } from '@angular/core';
    import { BrowserModule } from '@angular/platform-browser';
    import { LetModule, PushModule, UnpatchEventsModule } from '@rx-angular/template';

    import { AppComponent } from './app.component';

    @NgModule({
      declarations: [
        AppComponent,
      ],
      imports: [
        BrowserModule,
        UnpatchEventsModule
      ],
      providers: [],
      bootstrap: [AppComponent]
    })
    export class AppModule { }
  `);

    const file = appTree.readContent('app.module.ts');

    expect(file).toMatchSnapshot();
  });

  it('should replace UnpatchEventsModule identifier', async () => {
    appTree = await setupTestFile(`
      import { NgModule } from '@angular/core';
      import { BrowserModule } from '@angular/platform-browser';
      import { LetModule, PushModule, UnpatchEventsModule } from '@rx-angular/template';

      import { AppComponent } from './app.component';

      @NgModule({
        declarations: [
          AppComponent,
        ],
        imports: [
          BrowserModule,
          UnpatchEventsModule
        ],
        providers: [],
        bootstrap: [AppComponent]
      })
      export class AppModule { }
  `);

    const file = appTree.readContent('app.module.ts');

    expect(file).toMatchSnapshot();
  });

  function setupTestFile(fileInput: string, filePath = './app.module.ts') {
    const runner = new SchematicTestRunner(
      '@rx-angular/template',
      path.join(__dirname, '../../../migration.json')
    );
    const tree = new UnitTestTree(Tree.empty());

    tree.create(filePath, fileInput);

    return runner.runSchematicAsync(`update-1.0.0-beta.29`, {}, tree).toPromise();
  }
});
