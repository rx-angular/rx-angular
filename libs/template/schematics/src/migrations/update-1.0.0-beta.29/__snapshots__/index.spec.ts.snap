// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template Migration 1.0.0-beta.29 should replace LetModule + LetDirective module specifier 1`] = `
"import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { AppComponent } from './app.component';
import { LetModule } from '@rx-angular/template/let';
import { LetDirective } from '@rx-angular/template/let';

@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule, LetModule],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
"
`;

exports[`Template Migration 1.0.0-beta.29 should replace PushModule + PushPipe module specifier 1`] = `
"import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { AppComponent } from './app.component';
import { PushModule } from '@rx-angular/template/push';
import { PushPipe } from '@rx-angular/template/push';

@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule, PushModule],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
"
`;

exports[`Template Migration 1.0.0-beta.29 should replace UnpatchEventsModule + UnpatchDirective module specifier 1`] = `
"import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { AppComponent } from './app.component';
import { UnpatchModule } from '@rx-angular/template/unpatch';
import { UnpatchDirective } from '@rx-angular/template/unpatch';

@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule, UnpatchModule],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
"
`;

exports[`Template Migration 1.0.0-beta.29 should replace UnpatchEventsModule identifier 1`] = `
"import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { AppComponent } from './app.component';
import { LetModule } from '@rx-angular/template/let';
import { PushModule } from '@rx-angular/template/push';
import { UnpatchModule } from '@rx-angular/template/unpatch';

@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule, UnpatchModule],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
"
`;

exports[`Template Migration 1.0.0-beta.29 should replace all module specifiers 1`] = `
"import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { AppComponent } from './app.component';
import { LetModule } from '@rx-angular/template/let';
import { PushModule } from '@rx-angular/template/push';
import { UnpatchModule } from '@rx-angular/template/unpatch';

@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule, UnpatchModule],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
"
`;
